src/index.js

```diff
@@ -1,7 +1,7 @@
 import katex from 'katex';
 
-const inlineStartRule = /(?<=\s|^)\${1,2}(?!\$)/;
-const inlineRule = /^(\${1,2})(?!\$)((?:\\.|[^\\\n])+?)(?<!\$)\1(?=[\s?!\.,:]|$)/;
+const inlineStartRule = /(\s|^)\${1,2}(?!\$)/;
+const inlineRule = /^(\${1,2})(?!\$)((?:(?:\\[^\$]|[^\\\n])*?)(?:(?:\\\$\s)+|\\n|[^\$\n]))\1(?=[\s?!\.,:]|$)/;
```

Comment:
Why is `\\n` a special case? That would also be matched by `[^\$\n]`

Reply:
I see, this actually fails if it ends with any escaped character like `$\g$`

Reply:
Also this will match a newline character after an escaped $. Newline characters should not be allowed anywhere in the inline rule